packages/fuel-indexer-lib/src/manifest.rs

```diff
@@ -46,6 +47,30 @@ impl ToString for Module {
     }
 }
 
+/// Represents contract IDs in a `Manifest` struct.
+///
+/// - `Single`: Represents an optional single contract ID as an `Option<String>`.
+/// - `Multiple`: Represents a vector of optional contract IDs as a `Vec<Option<String>>`.
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub enum ContractIds {
+    Single(Option<String>),
+    Multiple(Vec<Option<String>>),
+}
+
+impl FromStr for ContractIds {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.starts_with('[') {
+            serde_json::from_str::<Vec<Option<String>>>(s)
+                .map(ContractIds::Multiple)
+                .map_err(|err| err.to_string())
```

Comment:
We're not really doing anything in this `.map_err`. Do we need it?

Reply:
If we remove `.map_err` the error type wont match the one required by the `FromStr` trait: `type Err = String;`

Reply:
@rvmelkonian So I don't see anything [in `FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) that means you _have_ to `.map_err` like this. But not a blocker for merge - can always be fixed in a follow up.

Thanks for the forbearance in all of this